src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -146,14 +148,17 @@ static ucs_status_t uct_rdmacm_cm_id_to_dev_addr(struct rdma_cm_id *cm_id,
         ucs_error("rdma_init_qp_attr (id=%p, qp_state=%d) failed: %m",
                   cm_id, qp_attr.qp_state);
         return UCS_ERR_IO_ERROR;
-     }
+    }
+
+    uct_rdmacm_cm_id_to_dev_name(cm_id, dev_name);
 
     if (ibv_query_port(cm_id->verbs, cm_id->port_num, &port_attr)) {
-        uct_rdmacm_cm_id_to_dev_name(cm_id, dev_name);
         ucs_error("ibv_query_port (%s) failed: %m", dev_name);
         return UCS_ERR_IO_ERROR;
     }
 
+    is_eth      = IBV_PORT_IS_LINK_LAYER_ETHERNET(&port_attr);
```

Comment:
is_eth variable seems redundant

Reply:
using it 2 times in this function. 
better to remove and use the macro every time?